src/rgw/rgw_op.h

```diff
@@ -180,6 +184,9 @@ class RGWOp : public DoutPrefixProvider {
   RGWQuotaInfo user_quota;
   int op_ret;
   int do_aws4_auth_completion();
+  #ifdef HAVE_JAEGER
+  std::unique_ptr<opentracing::Span> trace;
+  #endif
```

Comment:
i think this span should belong to `struct req_state` instead of `RGWOp`. there are some requests that start as one kind of op, but switch to another with `RGWHandler::retarget()`

`process_request()` is where we construct the `req_state`, and is probably the best place to start the trace

Reply:
ok. that would make the tracing related code concentrated in one place.
i guess the different ops will have only ops specific code in them (e.g. multipart upload id)

Reply:
done in 2947371
now we add those generic tags : `bucket_name`, `user_id`, `object_name`, `op` and `return`
and `upload_id` in `RGWInitMultipart` , `RGWCompleteMultipart` and in 'RGWPutObj` in case it's in multipart mode.